### PR TITLE
ocaml-sqlite3: 2.0.9 -> 5.0.1

### PR DESCRIPTION
**Pull request by kazcw**
###### Motivation for this change
version bump

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @massegi @vbgl

### CODE REVIEW

File: pkgs/development/ocaml-modules/sqlite3/default.nix [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
+  buildPhase = ''
+    runHook preBuild
+    dune build -p sqlite3
+    runHook postBuild
+  '';
+  installPhase = ''
+    runHook preInstall
+    ${opaline}/bin/opaline -prefix $out -libdir $OCAMLFIND_DESTDIR
+    runHook postInstall
+  '';
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/sqlite3/default.nix]: What about using `buildDunePackage`?

Comment (kazcw) [reply pkgs/development/ocaml-modules/sqlite3/default.nix]: `buildDunePackage` passes the `pname` given on to dune, but the `pname` of this package (`ocaml-sqlite3`) doesn't match what dune needs to hear (`sqlite3`). I don't know nix that well so I just copied some bits from `buildDune2Package` and modified them to give dune a different value from the `pname`. There might be a better way to do it?

Comment (vbgl) [reply pkgs/development/ocaml-modules/sqlite3/default.nix]: This sounds like the `pname` is wrong. You may just fix it ;-)

Comment (kazcw) [reply pkgs/development/ocaml-modules/sqlite3/default.nix]: Ah, I was thinking the `pname` was affecting the name of the package in nix.

File: pkgs/development/ocaml-modules/sqlite3/default.nix [hunk 2]
```diff
@@ -1,18 +1,16 @@
-{ stdenv, fetchurl, sqlite, ocaml, findlib, ocamlbuild, pkgconfig }:
+{ stdenv, fetchurl, sqlite, ocaml, pkgconfig, dune-configurator, buildDune2Package }:
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/sqlite3/default.nix]:
```suggestion
{ lib, fetchurl, sqlite, pkgconfig, buildDunePackage }:
```

File: pkgs/development/ocaml-modules/sqlite3/default.nix [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
 
-stdenv.mkDerivation rec {
-  pname = "ocaml-sqlite3";
-  version = "2.0.9";
+buildDune2Package rec {
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/sqlite3/default.nix]:
```suggestion
buildDunePackage rec {
```

File: pkgs/development/ocaml-modules/sqlite3/default.nix [hunk 4]
```diff
@@ -1,18 +1,16 @@
...
-    sha256 = "0rwsx1nfa3xqmbygim2qx45jqm1gwf08m70wmcwkx50f1qk3l551";
+    url = "https://github.com/mmottl/sqlite3-ocaml/releases/download/${version}/sqlite3-${version}.tbz";
+    sha256 = "0iymkszrs6qwak0vadfzc8yd8jfwn06zl08ggb4jr2mgk2c8mmsn";
   };
 
   nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ ocaml findlib ocamlbuild sqlite ];
-
-  createFindlibDestdir = true;
+  buildInputs = [ dune-configurator ocaml sqlite ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/sqlite3/default.nix]:
```suggestion
  buildInputs = [ sqlite ];
```

File: pkgs/development/ocaml-modules/sqlite3/default.nix [hunk 5]
```diff
@@ -1,18 +1,16 @@
...
 
   meta = with stdenv.lib; {
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/sqlite3/default.nix]:
```suggestion
  meta = with lib; {
```

**Review by vbgl — CHANGES_REQUESTED** [review]
This does not build.

Also please remove `meta.platforms`.

**Comment by kazcw** [comment]
My testing methodology has been insufficient to catch the build failure. I've been running `nix-shell -I nixpkgs=$NIXGIT ocamlPackages_latest.ocaml_sqlite3`,  `nix-shell -I nixpkgs=$NIXGIT ocamlPackages.ocaml_sqlite3`, and depending on it in the .nix for a project that accesses a database using the 5.x api.

**Comment by kazcw** [comment]
I've made the suggested improvements

File: pkgs/development/ocaml-modules/sqlite3/default.nix [hunk 6]
```diff
@@ -1,24 +1,21 @@
-{ stdenv, fetchurl, sqlite, ocaml, findlib, ocamlbuild, pkgconfig }:
+{ lib, fetchurl, sqlite, pkgconfig, buildDunePackage }:
 
-stdenv.mkDerivation rec {
-  pname = "ocaml-sqlite3";
-  version = "2.0.9";
+buildDunePackage rec {
+  pname = "sqlite3";
+  version = "5.0.1";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/sqlite3/default.nix]:
```suggestion
  minimumOCamlVersion = "4.05";

```

**Review by vbgl — APPROVED** [review]

**Pull request merged**